lib/sensu-handler.rb

```diff
@@ -126,14 +136,15 @@ def api_request(method, path, &_blk)
       if api_settings.nil?
         raise 'api.json settings not found.'
       end
-      domain = api_settings['host'].start_with?('http') ? api_settings['host'] : 'http://' + api_settings['host']
-      uri = URI("#{domain}:#{api_settings['port']}#{path}")
-      req = net_http_req_class(method).new(uri.path)
+      use_ssl = api_settings['ssl'].is_a?(Hash) ||
+                api_settings['host'].start_with?('https')
+      hostname = api_settings['host'].gsub(/https?:\/\//, '')
```

Comment:
A couple things:
- when not actually replacing chars just removing chars it is faster to use `tr` over `gsub`
- another option would be to split on `://` then you don't need to do any regex matching at all

Reply:
I don't think `tr` can handle the optional `s`. Splitting is an option, although maybe it's less explicit?

Reply:
If you pass it with `//` it will throw a type error but I believe it will convert the string you pass it to regex.

This should work:
```
irb(main):005:0> url = 'https://domain.tld'
=> "https://domain.tld"
irb(main):006:0> url.tr('https?', '')
=> "://domain.ld"
irb(main):007:0> url = 'http://domain.tld'
=> "http://domain.tld"
irb(main):008:0> url.tr('https?', '')
=> "://domain.ld"
```

Reply:
ok actually never mind I did not read the man pages well enough it is dangerous to do what I proposed as this will also get caught:
```
irb(main):001:0> url_http = 'http://sub.domain.tld'
=> "http://sub.domain.tld"
irb(main):004:0> url_http.tr('https?://', '')
=> "ub.domain.ld"
``